packages/position/src/react/index.js

```diff
@@ -4,84 +4,152 @@ import ReactDOM from 'react-dom'
+  return <Position position={positionFns.above} ref={ref} {...props} />
+})
 Above.displayName = 'Above'
 
-export function AboveLeft(props) {
-  return <Position position={positionFns.aboveLeft} {...props} />
-}
+export const AboveLeft = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.aboveLeft} ref={ref} {...props} />
+})
 AboveLeft.displayName = 'AboveLeft'
 
-export function AboveRight(props) {
-  return <Position position={positionFns.aboveRight} {...props} />
-}
+export const AboveRight = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.aboveRight} ref={ref} {...props} />
+})
 AboveRight.displayName = 'AboveRight'
 
-export function Below(props) {
-  return <Position position={positionFns.below} {...props} />
-}
+export const Below = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.below} ref={ref} {...props} />
+})
 Below.displayName = 'Below'
 
-export function BelowLeft(props) {
-  return <Position position={positionFns.belowLeft} {...props} />
-}
+export const BelowLeft = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.belowLeft} ref={ref} {...props} />
+})
 BelowLeft.displayName = 'BelowLeft'
 
-export function BelowRight(props) {
-  return <Position position={positionFns.belowRight} {...props} />
-}
+export const BelowRight = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.belowRight} ref={ref} {...props} />
+})
 BelowRight.displayName = 'BelowRight'
 
-export function RightOf(props) {
-  return <Position position={positionFns.rightOf} {...props} />
-}
+export const RightOf = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.rightOf} ref={ref} {...props} />
+})
 RightOf.displayName = 'RightOf'
 
-export function LeftOf(props) {
-  return <Position position={positionFns.leftOf} {...props} />
-}
+export const LeftOf = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.leftOf} ref={ref} {...props} />
+})
 LeftOf.displayName = 'LeftOf'
 
-function Position(props) {
-  const targetRef = React.useRef()
-  const showRef = React.useRef()
-  const child = React.Children.only(props.children)
+const Position = React.forwardRef((props, forwardedRef) => {
   const [style, setStyle] = React.useState({ position: 'absolute' })
 
-  React.useLayoutEffect(() => {
-    if (props.when && targetRef.current && showRef.current)
-      setStyle(props.position(targetRef.current).styleFor(showRef.current))
-  }, [props.when, targetRef, showRef, props])
-
-  const show =
-    props.when &&
-    React.cloneElement(props.show, {
-      ref: showRef,
-      style: {
-        ...child.props.style,
-        ...style
-      }
-    })
+  const ref = React.useRef()
+  React.useImperativeHandle(forwardedRef, () => ref.current)
+
+  const child = React.Children.only(props.children)
+  const tetheredRef = React.useRef()
+  const tetheredEl = React.cloneElement(props.show, {
```

Comment:
naming things is hard :(

i changed it because the word `show` seemed like it should be boolean and was tripping me up. `tether` comes from similar libraries and code i've written in the past. i'm open to changing it to something else...

Reply:
Yeah, the similarity to other libs makes sense.  Now there's just 2 names for the same thing in this lib.  `show` is not a boolean in this lib.  Probably not a big deal.  I love to use the same name for the same/related concept everywhere when possible.  Nbd at this point.

To clarify the api design, it was meant to read something like "Position below left of this thing (child), and *show* this other thing *when* something toggles true." fwiw